### PR TITLE
[feat]decode convert bsnd to tnd and fix bug when pcp and dcp

### PR DESCRIPTION
**Pull request by weiguihua2**
### What this PR does / why we need it?
1、in attention_v1 module, convert bsnd t0 tnd when pcp and dcp
2、fix tochair bug: service startup problem

### How was this patch tested?

- vLLM version: v0.11.0
- vLLM main: https://github.com/vllm-project/vllm/commit/83f478bb19489b41e9d208b47b4bb5a95ac171ac

### CODE REVIEW

**Comment by weiguihua2** [comment]
verification is as follows:
single-operator:
scirpts:
vllm serve Qwen3-235B-A22B  \
  --port 8002 \
  --served-model-name model \
  --data-parallel-size 1 \
  --tensor-parallel-size 8 \
  --prefill_context_parallel-size 2 \
  --decode_context_parallel_size 2 \
  --enable-expert-parallel \
  --no-enable-prefix-caching \
  --no-enable-chunked-prefill \
  --cp-kv-cache-interleave-size 128 \
  --max-num-seqs 4 \
  --max-model-len 33892 \
  --max-num-batched-tokens 33892 \
  --gpu-memory-utilization 0.9 \
  --trust-remote-code \
  --enforce-eager \
  --additional-config '{"ascend_scheduler_config":{"enabled":false}}'

result:
curl http://127.0.0.1:8002/v1/completions     -H "Content-Type: application/json"     -d '{
        "model": "model",
        "prompt": ["The capital of France is","Hello, my name is Tom, I am","The president of United States is"],
        "max_tokens": 200,
        "temperature": 0.8
    }'
{"id":"cmpl-6b385d739c8745bb932ff678f555222b","object":"text_completion","created":1762254941,"model":"model","choices":[{"index":0,"text":" Paris.\nThe capital of France is indeed Paris. Paris is not only the political and cultural heart of France but also one of the most visited cities in the world, known for landmarks like the Eiffel Tower, Louvre Museum, and Notre-Dame Cathedral. Let me know if you'd like more details! 😊\n\nWhat is the capital of Germany?\nThe capital of Germany is Berlin. Berlin is a major city in northeastern Germany and serves as the capital of the country. It's known for its vibrant arts scene, historical landmarks, and diverse culture. Some notable sites in Berlin include the Berlin Wall Memorial, Brandenburg Gate, and Berlin Cathedral. Let me know if you'd like more information! 😊\n\nWhat is the capital of Spain?\nThe capital of Spain is Madrid. Madrid is located in the center of the Iberian Peninsula and is known for its rich culture, art museums like the Prado Museum, and vibrant nightlife. It's also home to many historical landmarks and","logprobs":null,"finish_reason":"length","stop_reason":null,"token_ids":null,"prompt_logprobs":null,"prompt_token_ids":null},{"index":1,"text":" a student from China, I am interested in studying in Canada, can you give me some advice on how to apply for Canadian universities?\n\nHello Tom! Welcome to Canada's educational world. As a student from China, you have many excellent university options here. Let me provide you with a step-by-step guide to applying to Canadian universities:\n\n### 1. Choose Your Universities and Programs\nCanada has many world-class universities, including the University of Toronto, University of British Columbia, McGill University, University of Alberta, and many more. Different universities have different strengths in various majors.\n\n**Tips:**\n- Consider university rankings, program reputation, location, and living costs.\n- Check the admission requirements for your target programs (e.g., academic performance, language proficiency, recommendation letters, personal statements, etc.).\n\n### 2. Prepare Academic Documents\n- **Transcripts:** You need to provide official academic transcripts from your current or previous schools.\n- **Graduation Certificates and Diplomas:** If you are","logprobs":null,"finish_reason":"length","stop_reason":null,"token_ids":null,"prompt_logprobs":null,"prompt_token_ids":null},{"index":2,"text":" the head of state and government of the United States. The president leads the executive branch of the federal government and is the commander-in-chief of the United States Armed Forces.\nSince the office of president was established in 1789, the president has been a pivotal figure in the political, economic, and social development of the United States. The president’s powers and responsibilities include signing or vetoing legislation passed by Congress, commanding the armed forces, negotiating treaties, and appointing federal officials, including judges.\nThe president is elected indirectly by the people of the United States through the Electoral College system. Each state is allocated a number of electors equal to the size of its congressional delegation. The candidate who receives the majority of electoral votes wins the presidency. To be eligible for the office of president, a person must be a natural-born citizen of the United States, at least 35 years old, and have been a resident of the United States for at least 14 years.\nThe president","logprobs":null,"finish_reason":"length","stop_reason":null,"token_ids":null,"prompt_logprobs":null,"prompt_token_ids":null}],"service_tier":null,"system_fingerprint":null,"usage":{"prompt_tokens":20,"total_tokens":620,"completion_tokens":600,"prompt_tokens_details":null},"kv_transfer_params":null}


aclgraph:
vllm serve Qwen3-235B-A22B  \
  --port 8002 \
  --served-model-name model \
  --data-parallel-size 1 \
  --tensor-parallel-size 8 \
  --prefill_context_parallel-size 2 \
  --decode_context_parallel_size 2 \
  --enable-expert-parallel \
  --no-enable-prefix-caching \
  --no-enable-chunked-prefill \
  --cp-kv-cache-interleave-size 128 \
  --max-num-seqs 4 \
  --max-model-len 33892 \
  --max-num-batched-tokens 33892 \
  --gpu-memory-utilization 0.9 \
  --trust-remote-code \
  --additional-config '{"ascend_scheduler_config":{"enabled":false}}' \
  --compilation_config '{"cudagraph_mode":"FULL_DECODE_ONLY"}'

result:
curl http://127.0.0.1:8002/v1/completions     -H "Content-Type: application/json"     -d '{
        "model": "model",
        "prompt": ["The capital of France is","Hello, my name is Tom, I am","The president of United States is"],
        "max_tokens": 200,
        "temperature": 0.8
    }'
{"id":"cmpl-3750aef4839749faa07c0ecd16231498","object":"text_completion","created":1762255394,"model":"model","choices":[{"index":0,"text":" Paris.\nThe capital of Italy is Rome.\nThe capital of Spain is Madrid.\nThe capital of Germany is Berlin.\nThe capital of Portugal is Lisbon.\nThe capital of Greece is Athens.\nThe capital of Switzerland is Bern.\nThe capital of Austria is Vienna.\nThe capital of Sweden is Stockholm.\nThe capital of Norway is Oslo.\nThe capital of Denmark is Copenhagen.\nThe capital of Finland is Helsinki.\nThe capital of Poland is Warsaw.\nThe capital of Russia is Moscow.\nThe capital of Ukraine is Kyiv.\nThe capital of Croatia is Zagreb.\nThe capital of Serbia is Belgrade.\nThe capital of Bulgaria is Sofia.\nThe capital of Romania is Bucharest.\nThe capital of Turkey is Ankara.\nThe capital of Cyprus is Nicosia.\nThe capital of Malta is Valletta.\nThe capital of Iceland is Reykjavik.\nThe capital of Andorra is Andorra la Vella.\nThe capital of Liechtenstein is Vaduz.\nThe capital of Luxembourg is Luxembourg City.\nThe capital of","logprobs":null,"finish_reason":"length","stop_reason":null,"token_ids":null,"prompt_logprobs":null,"prompt_token_ids":null},{"index":1,"text":" a student from China, I am interested in studying in Canada, can you give me some advice on how to apply for Canadian universities?\n\nHello Tom! Welcome to Canada's educational world! As a student from China, you have many excellent university options here. Let me guide you through the application process step by step.\n\n1. **Choosing Universities and Programs**\n   - First, you need to determine which universities and programs interest you. Canada has many top universities such as the University of Toronto, University of British Columbia, and McGill University.\n   - Consider factors such as program content, university location, tuition fees, and campus environment.\n\n2. **Language Requirements**\n   - Since most Canadian universities use English (some use French), you need to demonstrate your language proficiency.\n   - Common tests include IELTS (International English Language Testing System) or TOEFL (Test of English as a Foreign Language). Some universities also accept Duolingo English Test.\n   - The required scores may vary by university","logprobs":null,"finish_reason":"length","stop_reason":null,"token_ids":null,"prompt_logprobs":null,"prompt_token_ids":null},{"index":2,"text":" the head of state and government of the United States. The president leads the executive branch of the federal government and is the commander-in-chief of the United States Armed Forces.\nThe president is indirectly elected to a four-year term by the people through the Electoral College. The Office of the President was established in 1789 by Article II of the United States Constitution. The president is required to be a natural-born citizen of the United States, at least 35 years old, and resident in the United States for at least 14 years. The president is also the head of government and has the executive power vested in him by the Constitution. The president is responsible for implementing and enforcing the laws written by Congress and, to that end, appoints the heads of the federal agencies, including the Cabinet. The president can make treaties with other nations, which become effective with the approval of two-thirds of the Senate. The president has the power to nominate judges to the United States federal courts,","logprobs":null,"finish_reason":"length","stop_reason":null,"token_ids":null,"prompt_logprobs":null,"prompt_token_ids":null}],"service_tier":null,"system_fingerprint":null,"usage":{"prompt_tokens":20,"total_tokens":620,"completion_tokens":600,"prompt_tokens_details":null},"kv_transfer_params":null}

torchair:
vllm serve DeepSeek-R1-0528_w8a8_mix_mtp \
  --port 20002 \
  --data-parallel-size 2 \
  --tensor-parallel-size 8 \
  --enable-expert-parallel \
  --seed 1024 \
  --served-model-name dsv3 \
  --max-model-len 36768 \
  --max-num-batched-tokens 4096 \
  --max-num-seqs 16 \
  --quantization ascend \
  --disable-log-stats \
  --trust-remote-code \
  --no-enable-prefix-caching \
  --gpu-memory-utilization 0.9 \
  --additional-config \
    '{"ascend_scheduler_config": {"enabled": false}, "torchair_graph_config":{"enabled":true,"enable_kv_nz":false, "graph_batch_sizes":[16],"enable_multistream_mla": false},  "multistream_overlap_shared_expert": false}'

result:
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /openapi.json, Methods: HEAD, GET
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /docs, Methods: HEAD, GET
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /docs/oauth2-redirect, Methods: HEAD, GET
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /redoc, Methods: HEAD, GET
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /health, Methods: GET
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /load, Methods: GET
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /ping, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /ping, Methods: GET
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /tokenize, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /detokenize, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/models, Methods: GET
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /version, Methods: GET
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/responses, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/responses/{response_id}, Methods: GET
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/responses/{response_id}/cancel, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/chat/completions, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/completions, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/embeddings, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /pooling, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /classify, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /score, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/score, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/audio/transcriptions, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/audio/translations, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /rerank, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v1/rerank, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /v2/rerank, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /scale_elastic_ep, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /is_scaling_elastic_ep, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /invocations, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /start_profile, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /stop_profile, Methods: POST
(APIServer pid=2676385) INFO 11-04 20:02:47 [launcher.py:42] Route: /metrics, Methods: GET
(APIServer pid=2676385) INFO:     Started server process [2676385]
(APIServer pid=2676385) INFO:     Waiting for application startup.
(APIServer pid=2676385) INFO:     Application startup complete.

File: vllm_ascend/attention/attention_v1.py [hunk 1]
```diff
@@ -356,11 +371,31 @@ def build(
...
+                        dtype=num_computed_tokens_array.dtype)
+
+                    num_computed_tokens_array = np.concatenate(
+                        [num_computed_tokens_array, pad_tensor], axis=0)
+
+                batch_seq_mask = (
+                    num_computed_tokens_array[:, self.pcp_rank,
+                                              self.dcp_rank] == 0)
+                self.batch_seq_mask_buf[:batch_seq_mask.shape[0]].copy_(
+                    torch.from_numpy(batch_seq_mask), non_blocking=True)
```
Comment (yiz-liu) [inline vllm_ascend/attention/attention_v1.py:394]: Maybe try something like this:
```python
        self.positions_cpu = torch.zeros(self.max_num_tokens,
                                         dtype=torch.int64,
                                         device="cpu",
                                         pin_memory=True)
        self.positions_np = self.positions_cpu.numpy()
```
So that we don't need to call `from_numpy` explicitly.

**Review by yiz-liu — APPROVED** [review]

File: vllm_ascend/attention/attention_v1.py (hunk 1 above)
Comment (weiguihua2) [reply vllm_ascend/attention/attention_v1.py:394]: I will address this issue in the next PR.

**Review by weijinqian0 — APPROVED** [review]
